XCUITests/AuthenticationTest.swift

```diff
@@ -228,104 +173,98 @@ class AuthenticationTest: BaseTestCase {
+        navigator.performAction(Action.SetPasscode)
+        setInterval("After 5 minutes")
 
         // Enter wrong passcode
-        navigator.goto(LoginsSettings)
+        navigator.goto(LockedLoginsSettings)
         waitforExistence(app.navigationBars["Enter Passcode"])
-        typePasscode("233777")
+
+        navigator.performAction(Action.LoginPasscodeTypeIncorrectOne)
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 2)."])
-        typePasscode("333777")
+
+        navigator.performAction(Action.LoginPasscodeTypeIncorrectOne)
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 1)."])
-        typePasscode("333777")
+
+        navigator.performAction(Action.LoginPasscodeTypeIncorrectOne)
         waitforExistence(app.staticTexts["Maximum attempts reached. Please try again later."])
     }
 
     func testWrongPasscodeAttemptsPersistAcrossEntryAndConfirmation() {
-        enablePasscode("133777")
+         navigator.performAction(Action.SetPasscode)
 
         // Enter wrong passcode on Logins
-        navigator.goto(LoginsSettings)
+        navigator.goto(LockedLoginsSettings)
         waitforExistence(app.navigationBars["Enter Passcode"])
-        typePasscode("233777")
+
+        navigator.performAction(Action.LoginPasscodeTypeIncorrectOne)
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 2)."])
 
         // Go back to Passcode, and enter a wrong passcode, notice the error count
         navigator.goto(PasscodeSettings)
-        app.staticTexts["Change Passcode"].tap()
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode("233777")
+        userState.passcode = "222222"
+        navigator.performAction(Action.ConfirmPasscodeToChangePasscode)
+
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 1)."])
         app.buttons["Cancel"].tap()
 
-        disablePasscode("133777")
+        userState.passcode = "111111"
+        navigator.nowAt(PasscodeSettings)
+        navigator.performAction(Action.DisablePasscode)
+        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"])
     }
 
     func testChangedPasswordMustBeNew() {
-        enablePasscode("133777")
-        app.staticTexts["Change Passcode"].tap()
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode("133777")
-        waitforExistence(app.staticTexts["Enter a new passcode"])
-        typePasscode("133777")
+        navigator.performAction(Action.SetPasscode)
+        userState.newPasscode = "111111"
+
+        navigator.performAction(Action.ChangePasscode)
         waitforExistence(app.staticTexts["New passcode must be different than existing code."])
         app.navigationBars["Change Passcode"].buttons["Cancel"].tap()
```

Comment:
Nit: Use `navigator.back()` instead of `tap()`.

Reply:
This does not seem to work on iPad, will open a bug for this and the other cases seen on these tests where that ´navigator.back()´ does not work